pythermalcomfort/charts/heatmap.py

```diff
@@ -0,0 +1,215 @@
+from math import ceil, floor
+import numpy as np
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+
+from pythermalcomfort.charts.theme import (
+    tight_margins,
+    index_mapping_dictionary,
+)
+
+
+def heatmap(df, var, global_local, si_ip, show_summary=False):
+    """Generates a heatmap with the desired variable.
+
+    Parameters
+    ----------
+    df : DataFrame
+        A DataFrame containing the input data. Must have the following columns:
+        'DateTime' (Date and time), 'Hour' (Hour of the day), 'Month Name' (Month name), 'Day' (Day of the month),
+        and the variable to be plotted.
+
+    var : str
+        The variable to be plotted.
+
+    global_local : str
+        The scope of the data. It can be 'global' or 'local'.
+
+    si_ip : str
+        The unit system to use. It can be 'si' or 'ip'.
+
+    show_summary : bool, optional
+        If True, a summary of the data is shown. Default is False.
+
+    Returns
+    -------
+    fig : graph_objects.Figure
+        Plotly figure object.
+
+    """
+
+    var_name = index_mapping_dictionary[var]["name"]
+    var_unit = index_mapping_dictionary[var][si_ip]["unit"]
+    var_range = index_mapping_dictionary[var][si_ip]["range"]
+    var_color = index_mapping_dictionary[var]["colors"]
+
+    if global_local == "global":
+        range_z = var_range
```

Comment:
I don't think we should be using this code, but instead we should allow the user to pass the min and max value. If they don't, then we can use the values from the data set.